core/chains/evm/evm_txm.go

```diff
@@ -36,19 +36,19 @@ func newEvmTxm(
 			"gasLimitDefault", cfg.EvmGasLimitDefault(),
 		)
 
+		checker := &txmgr.CheckerFactory{Client: client}
+
 		// build estimator from factory
 		estimator := gas.NewEstimator(lggr, client, cfg)
 
-		// build forwarder manager for evm txm
-		var fwdMgr *forwarders.FwdMgr
 		if cfg.EvmUseForwarders() {
-			fwdMgr = forwarders.NewFwdMgr(db, client, logPoller, lggr, cfg)
+			fwdMgr := forwarders.NewFwdMgr(db, client, logPoller, lggr, cfg)
+			txmgr.NewTxm(db, client, cfg, opts.KeyStore, opts.EventBroadcaster, lggr, checker, estimator, fwdMgr)
 		} else {
 			lggr.Info("EvmForwarderManager: Disabled")
+			txm = txmgr.NewTxm(db, client, cfg, opts.KeyStore, opts.EventBroadcaster, lggr, checker, estimator, nil)
```

Comment:
I was getting errors earlier because when *forwarders.FwdMgr is nil it still confirms to the txmgrtypes.Forwarder interface (since Golang supports nil receivers) at compile time but you'll get a runtime error due to nil pointer dereferencing when forwarder calls Start() an Close() (here's a toy example of what I'm talking about https://gotipplay.golang.org/p/nJH20Yy0LRz).

I'm not sure what best practice is regarding nil receivers. I read it's commonplace to do nil checks within the struct that implements the interface. Semantically though, I don't believe it makes sense to call Start() or Close() on a nil receiver and it doesn't seem like we do nil receiver checks for other interfaces so I don't want to rock the boat. The convention in this repo seems to be in to pass in nil interfaces and avoid passing in nil pointers. It feels a bit dangerous though because someone can come along and pass in a nil pointer by accident, especially if the parameter is optional).

Open to changing this to add nil checks in the evm forward manager struct. I also played around with reflection as a possibility (in txmgr.go adding checking for `v := reflect.ValueOf(b.fwdMgr) and v.IsNil()` .

Reply:
could you use to keep from repeating the `NewTxm` line?
```go
fwdMgr ForwarderManager[...] := nil
if {...}
else {...}
txm = txmgr.NewTxm(...)
```